### DISCUSSION TITLE
hello ,how I can create custom certificate extension in java CSR request for ejbca so that it can reflect on certificate.pem,

### DISCUSSION THREAD

**Discussion by Pooja27-proj** [Keyfactor/ejbca-ce#669] [Q&A] [answered]
I am trying to add below in certificate by msgbuilder.addExtension(new ASN1ObjectIdentifier("1.3.6.1.5.5.7.1.34"),false,nfTypes); but its not reflecting in certificate response.what changes should i do?
[3] {
    SEQUENCE {
     SEQUENCE {
:      OBJECT IDENTIFIER nfTypes (1 3 6 1 5 5 7 1 34)
     OCTET STRING, encapsulates {
       SEQUENCE {
:        IA5String 'AMF'
      :         }
      :        }
      :       }

**Comment by primetomas** [comment, marked as answer]
You can add custom extensions in many ways. One is to use "Custom extensions" in EJBCA.
https://docs.keyfactor.com/ejbca/latest/custom-certificate-extensions

Another is to use "Allow Extension Override" in the certificate profile, to copy extensions from your request into the issued certificate, if you prefer to do that.

Nothing like that is copied from the CSR by default, without you explicitly allowing it. That would be an insecure default setting.

**Reply by Pooja27-proj** [reply]
Thank you so much for your reply,I have tried this "Custom extensions" in EJBCA.
https://docs.keyfactor.com/ejbca/latest/custom-certificate-extensions it  is working .let me try same by request

**Comment by Pooja27-proj** [comment]
hi ,is there any java code available for Revoke request,?

**Reply by primetomas** [reply]
This is a different topic that needs a different thread. I will close this discussion, please create a new one.
